feature_details.md

```diff
@@ -0,0 +1,417 @@
+
+There are 3 categories of use-cases targeted by this feature which have different motivations and constraints :
+
+### Single Page Apps (SPA)
+The transitions targeted by this feature are in theory already possible for SPAs today but are complicated to get right. In order to achieve the same effect, the developer needs to do the following:
+
+* Retain the current version of the DOM
+* Asynchronously set up the new version of the DOM
+* Set up animations between live DOM elements for both states
+* Clean up the previous state at the end of the transition.
+
+Since live DOM needs to be present for both states at the same time, it's common to get the details wrong (eg, previous state might remain clickable or may stick around after the transition). This leads to difficult-to-debug issues and can have a negative impact on accessibility.
+
+We want to make authoring such transitions easy with a native API that offers some level of customization.
+
+### Multi Page Apps (MPA)
+It’s not possible to have these transitions today for multi-page apps (or same-origin navigations). In fact, a lack of this ends up being one of the drivers for web authors to create SPAs. We want this API to enable authoring transitions within an MPA with a similar feature set as the SPA case above. 
+
+### Cross Origin Transitions
+Cross-origin transitions are infeasible similar to MPAs. A common use-case for this is content aggregator sites (search engines, social media sites, news aggregators, etc.). These sites frequently display a hero image/header which could be animated during the navigation to provide a seamless transition.
+
+Cross-origin transitions have additional security/privacy constraints. In particular we must ensure the feature does not allow any cross-site information sharing which imposes the following restrictions :
+
+* The animations associated with the transition need to be defined completely by the previous Document, which is aware of the site the user is navigating to and has contextual information about the user journey.
+
+* The animations need to be executed such that there are no observable side-effects for the new and previous Documents. For example, setting an obscure animation for a specific element on the incoming Document could be used to exchange information between origins. Enforcing no side-effects is an important consideration, since it can influence the stage in the rendering pipeline where the UA executes these animations.
+
+An important assumption in the design of this feature is that it's reasonable to limit customization options for cross-origin transitions to make it easier for UAs to enforce the constraints above. But the API/implementation should strive to not unnecessarily limit capabilities for same-origin transitions due to cross-origin constraints.
+
+## Sample Code
+This section explains how the feature can be used to add transitions to MPAs. There are a set of use-cases and edge cases to consider :
+
+### Root Transition
```

Comment:
it'll be nice to code snippet and explanation side by side.

Reply:
Finding it hard to put a mult-line code snippet next to a description with a table.